pkg/smb/controllerserver.go

```diff
@@ -223,6 +224,14 @@ func (d *Driver) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequest)
 				return nil, status.Errorf(codes.Internal, "archive subdirectory(%s, %s) failed with %v", internalVolumePath, archivedInternalVolumePath, err.Error())
 			}
 		} else {
+			if _, err2 := os.Lstat(internalVolumePath); err2 == nil {
```

Comment:
Would it be better to pass `-o dir_mode=0777,file_mode=0777` in `internalMount()` above? It would fix any potential `RemoveAll` in the archive too and who knows what else.

Reply:
Makes sense, thanks for pointing out!

Reply:
I made a fix here: https://github.com/kubernetes-csi/csi-driver-smb/pull/843